queue/util/message-parser.js

```diff
@@ -28,5 +28,8 @@ exports.decode = function (message) {
     throw new Error(`Error decoding event. Errors: "${errors.join(',')}"`)
   }
 
+  // Convert the event's "data" property back to its orignal model type
+  event.convertData()
```

Comment:
The only functional change

Reply:
Might be worth checking for errors here.

Reply:
@Druotic and I were talking and basically we were thinking that since this data will be produced by a producer (and presumably wrapped), we shouldn't have to handle errors here. Thoughts?

Reply:
As long as the fashion model errors are descriptive enough I think handling them down the line is fine.

Reply:
>@Druotic and I were talking and basically we were thinking that since this data will be produced by a producer (and presumably wrapped), we shouldn't have to handle errors here. Thoughts?

We shouldn't ever see errors here, but I don't see any harm in passing in an error array and logging any errors if they do start happening for some reason. Could help with debugging in the future if someone breaks something accidentally.